lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -139,23 +139,39 @@ def safe_make_column_not_nullable(table, column)
     end
 
     validated_table = PgHaMigrations::Table.from_table_name(table)
-    tmp_constraint_name = "tmp_not_null_constraint_#{OpenSSL::Digest::SHA256.hexdigest(column.to_s).first(7)}"
+    quoted_column_name = connection.quote_column_name(column)
+    column_str = column.to_s
 
-    if validated_table.check_constraints.any? { |c| c.name == tmp_constraint_name }
-      raise PgHaMigrations::InvalidMigrationError, "A constraint #{tmp_constraint_name.inspect} already exists. " \
-        "This implies that a previous invocation of this method failed and left behind a temporary constraint. " \
-        "Please drop the constraint before attempting to run this method again."
-    end
+    # First, look for existing constraints that match the IS NOT NULL pattern for this column
+    existing_constraint = validated_table.check_constraints.select do |c|
+      c.definition =~ /\ACHECK \(*(#{Regexp.escape(column_str)}|#{Regexp.escape(quoted_column_name)}) IS NOT NULL\)*\Z/i
```

Comment:
I did just approve this but it just occurred to me that maybe we want specific tests for this regex? Like, it looks right to me, but should we write a test that makes sure this method doesn't discover a constraint that has a `IS NOT NULL AND ...` kind of definition?

Reply:
Yeah, I think we have those tests for `safe_make_column_not_nullable_from_check_constraint` already; I'll add those before merging.